File: 백준/Silver/10431. 줄세우기/줄세우기.py
import sys

p = int(input())

for _ in range(p):
    class_num = list(map(int, input().split()))
    t = class_num[0]
    children = []
    for i in range(20):
        s = class_num[i+1]
        children.append(s)
    res = [children[0]]
    cnt = 0
    for i in range(1, 20):
        res.append(children[i])
        idx = 0
        for j in range(i):
            if children[j] > children[i]:
                new_res = res[:j-1] + [children[i]] + res[j:19]
                res = new_res
                cnt += 1

    print(t, cnt)

File: 백준/Silver/10431. 줄세우기/줄세우기.java
import java.io.*;
import java.util.*;

public class Main{
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int p = Integer.parseInt(br.readLine());

        for (int tc = 0; tc < p; tc ++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int t = Integer.parseInt(st.nextToken());

            int[] children = new int[20];
            for(int i = 0; i < 20; i++) {
                children[i] = Integer.parseInt(st.nextToken());
            }

            List<Integer> res = new ArrayList<>();
            res.add(children[0]);
            int cnt = 0;
            for (int i = 1; i < 20; i++) {
                res.add(children[i]);
                for (int j=0; j< i; j++) {
                    if (children[j] > children[i]) {
                        res.remove(i);
                        res.add(j, children[i]);
                        cnt++;
                   
                    }
                }
            }
            System.out.println(t + " " + cnt);

        }
    }
}